File: c++/Q1.cpp
#include <iostream>
#include <string>

using namespace std;

class Pessoa{
    private:
    string nome;
    string nacionalidade;
    string posicao;
    int dataNascimento;
    float peso;
    float altura; 
    
    
    public:
        Pessoa(string posicao, string nome, int dataNascimento, float peso, float alturam, string nacionalidade) : 
        nome(nome), dataNascimento(dataNascimento), peso(peso), altura(altura), nacionalidade(nacionalidade), posicao(posicao){}

        virtual void exibirInformacoes(){
            cout << "Nome: " << nome << "Data de Nascimento: " << dataNascimento << "peso: " << peso << "altura: " << altura << endl;
        }
        string getPosicao(){
            return posicao;
        }
        string getNacionalidade(){
            return nacionalidade;
        }
        string getNome(){
            return nome;
        }
        int getdataNascimento(){
            return dataNascimento;
        }
        float getPeso(){
            return peso;
        }
        float getAltura(){
            return altura;
        }

        void setNome(string newNome){
            nome = newNome;
        }
        void setdataNasciment(int newData){
            dataNascimento = newData;
        }
        void setPeso(float newPeso){
            peso = newPeso;
        }
        void setAltura(float newAltura){
            altura = newAltura;
        }     
        void setNacionalidade(string newNacao){
            nacionalidade = newNacao;
        }
        void setPosica(string newPosicao){
            posicao = newPosicao;
        }
};

class Jogador : public Pessoa{
    private:
        

    public:
     
        void calcularIdade(){

        } 
};

File: myenv/src/Q1.py
import datetime
from abc import ABC, abstractmethod

class Pessoa(ABC):
    def __init__(self, nome, dataNascimento, altura, peso ):
        self.nome = nome
        self.dataNascimento = dataNascimento
        self.altura = altura
        self.peso = peso

    def toString(self):
        print(f'''
            +-----------------------------------------+
            |       DADOS DO JOGADOR          |
            +-----------------------------------------+
             Nome do Jogador: {self.getNome()}    
             Posicao: {self.getPosicao()}            
             Altura: {self.getAltura()}               
             Peso: {self.getPeso()}               
             Data de Nascimento:{self.getDatanascimento()}
             Nacionalidade: {self.getNacionalidade()}     
            +-----------------------------------------+
            ''')

class Jogador(Pessoa):
    def __init__(self,nome, dataNascimento, altura, peso):
        super().__init__(nome, dataNascimento, altura, peso)
        
    def toString(self):
        return super().toString()

    def calcular_tempo(self):
        data_atual =  datetime.datetime.now()
        self.dataNascimento = datetime.datetime.strptime(self.dataNascimento, "%Y-%m-%d")

        diferenca = data_atual - self.dataNascimento
        self.idade = diferenca.days // 365

        print(f"\nA idade do jogador {self.nome} é: {self.idade}")

        if self.idade >= 40:
            print("Jogador aposentado!")        
        elif self.idade >= 36 or self.idade <= 39:
            print("Jogador meio-camp")
        elif self.idade <= 35:
            print("Jogador Atacante")

